server/src/test/java/org/elasticsearch/common/util/MapsTests.java

```diff
@@ -123,6 +128,56 @@ public void testDeepEquals() {
         assertFalse(Maps.deepEquals(map, mapModified));
     }
 
+    public void testCollectToUnmodifiableSortedMap() {
+        SortedMap<String, String> canadianProvinces = Stream.of(
+                new Tuple<>("ON", "Ontario"),
+                new Tuple<>("QC", "Quebec"),
+                new Tuple<>("NS", "Nova Scotia"),
+                new Tuple<>("NB", "New Brunswick"),
+                new Tuple<>("MB", "Manitoba"))
+            .collect(Maps.toUnmodifiableSortedMap(Tuple::v1, Tuple::v2));
+
+        assertThat(canadianProvinces, equalTo(new TreeMap<>(Maps.ofEntries(List.of(
+            entry("ON", "Ontario"),
+            entry("QC", "Quebec"),
+            entry("NS", "Nova Scotia"),
+            entry("NB", "New Brunswick"),
+            entry("MB", "Manitoba"))
+        ))));
+        expectThrows(UnsupportedOperationException.class, () -> canadianProvinces.put("BC", "British Columbia"));
+    }
+
+    public void testCollectRandomListToUnmodifiableSortedMap() {
+        List<Tuple<String, String>> tuples = randomList(0, 100,
+            () -> new Tuple<>(randomAlphaOfLength(10), randomAlphaOfLength(10)))
+            .stream()
+            .distinct()
+            .collect(Collectors.toList());
+        Collections.shuffle(tuples, Randomness.get());
```

Comment:
Slightly shorter variant:

```suggestion
        Randomness.shuffle(tuples);
```